### PR TITLE
[COST-3123] ocp on aws api cost type

### PR DESCRIPTION
**Pull request by adberglund**
## Jira Ticket

[COST-3123](https://issues.redhat.com/browse/COST-3123)

## Description

This will allow the cost_type param to be used on /api/cost-management/v1/reports/openshift/infrastructures/aws/ endpoints and will pull the underlying db field for cost usage based on that term. 

## Testing

1. Checkout Branch
2. `make create-test-customer; make load-test-customer-data`
3. WAIT
4. Try http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/costs/?filter[time_scope_value]=-1&cost_type=unblended_cost
5. Try same api wiht cost_type=blended_cost and cost_type=savingsplan_effective_cost
6. Note until #4030 merges, values will be 0

## Notes

...

### CODE REVIEW

File: koku/api/report/aws/openshift/query_handler.py [hunk 1]
```diff
@@ -129,7 +129,11 @@ def __init__(self, parameters):
             parameters    (QueryParameters): parameter object for query
 
         """
-        self._mapper = OCPAWSProviderMap(provider=self.provider, report_type=parameters.report_type)
+        self._mapper = OCPAWSProviderMap(
+            provider=self.provider,
+            report_type=parameters.report_type,
+            cost_type=parameters.parameters.get("cost_type"),
```
Comment (maskarb) [inline koku/api/report/aws/openshift/query_handler.py:135]: It probably makes sense to add a `cost_type` property to QueryParameters like we do [here](https://github.com/project-koku/koku/blob/main/koku/api/query_params.py#L463-L466), but not a super critical change.

**Review by maskarb — APPROVED** [review]
lgtm!

**Comment by adberglund** [comment]
https://ci.ext.devshift.net/job/project-koku-koku-pr-check/7596/ failures are from another recent change around tag filtering on wildcard.

**Review by maskarb — APPROVED** [review]

**Review by maskarb — APPROVED** [review]
lgtm!

**Pull request merged**